app/client/components/identity/identity.ts

```diff
@@ -0,0 +1,73 @@
+import * as ConsentsAPI from "./idapi/consents";
+import * as NewslettersAPI from "./idapi/newsletters";
+import * as NewslettersSubscriptionsAPI from "./idapi/newsletterSubscriptions";
+import * as RemoveSubscriptionsAPI from "./idapi/removeSubscriptions";
+import * as UserAPI from "./idapi/user";
+
+import {
+  ConsentOption,
+  ConsentOptionCollection,
+  ConsentOptionType
+} from "./models";
+
+const isNewsletter = (option: ConsentOption): boolean =>
+  option.type === ConsentOptionType.NEWSLETTER;
+const isConsent = (option: ConsentOption): boolean =>
+  option.type !== ConsentOptionType.NEWSLETTER;
+
+const mapSubscriptions = (
+  subscriptions: string[],
+  options: ConsentOption[]
+): ConsentOption[] =>
+  options.map(option => ({
+    ...option,
+    subscribed: option.subscribed ? true : subscriptions.includes(option.id)
+  }));
+
```

Comment:
Could we get some inline comments here describing what is going on with the data from each response. And an example of the response from each api would be good.
Also are there tests covering this, it feels like it would be hard to make changes to this logic as there are 4 data models, and tests would help in this.